k4GaudiPandora/src/DDPandoraPFANewAlgorithm.cc

```diff
@@ -0,0 +1,489 @@
+}
+*/
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+pandora::StatusCode DDPandoraPFANewAlgorithm::RegisterUserComponents() const {
+  PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, LCContent::RegisterAlgorithms(*m_pPandora));
+  PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, LCContent::RegisterBasicPlugins(*m_pPandora));
+
+  if (m_settings.m_useDD4hepField) {
+    dd4hep::Detector& mainDetector = dd4hep::Detector::getInstance();
+
+    PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+                             PandoraApi::SetBFieldPlugin(*m_pPandora, new DDBFieldPlugin(mainDetector)));
+  } else {
+    PANDORA_RETURN_RESULT_IF(
+        pandora::STATUS_CODE_SUCCESS, !=,
+        LCContent::RegisterBFieldPlugin(*m_pPandora, m_settings.m_innerBField, m_settings.m_muonBarrelBField,
+                                        m_settings.m_muonEndCapBField));
+  }
+
+  PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+                           LCContent::RegisterNonLinearityEnergyCorrection(
+                               *m_pPandora, "NonLinearity", pandora::HADRONIC, m_settings.m_inputEnergyCorrectionPoints,
+                               m_settings.m_outputEnergyCorrectionPoints));
+
+  lc_content::LCSoftwareCompensationParameters softwareCompensationParameters;
+  softwareCompensationParameters.m_softCompParameters              = m_settings.m_softCompParameters;
+  softwareCompensationParameters.m_softCompEnergyDensityBins       = m_settings.m_softCompEnergyDensityBins;
+  softwareCompensationParameters.m_energyDensityFinalBin           = m_settings.m_energyDensityFinalBin;
+  softwareCompensationParameters.m_maxClusterEnergyToApplySoftComp = m_settings.m_maxClusterEnergyToApplySoftComp;
+  softwareCompensationParameters.m_minCleanHitEnergy               = m_settings.m_minCleanHitEnergy;
+  softwareCompensationParameters.m_minCleanHitEnergyFraction       = m_settings.m_minCleanHitEnergyFraction;
+  softwareCompensationParameters.m_minCleanCorrectedHitEnergy      = m_settings.m_minCleanCorrectedHitEnergy;
+
+  PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+                           LCContent::RegisterSoftwareCompensationEnergyCorrection(*m_pPandora, "SoftwareCompensation",
+                                                                                   softwareCompensationParameters));
+
+  return pandora::STATUS_CODE_SUCCESS;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void DDPandoraPFANewAlgorithm::FinaliseSteeringParameters() {
+  // ATTN: This function seems to be necessary for operations that cannot easily be performed at construction of the processor,
+  // when the steering file is parsed e.g. the call to GEAR to get the inner bfield
+  MsgStream log(msgSvc(), name());
```

Comment:
Simply marking this to not forget about it, but this will potentially be gone with the rework I proposed.